shared/constants/types/settings.js

```diff
@@ -15,49 +16,48 @@ export type PendingInvite = {
 
 export type AcceptedInvite = {
   username: string,
-  fullname: string,
-  currentlyFollowing: boolean,
 } & InviteBase
 
 export type Invitation = {
   created: number,
-  email: string,
+  email: ?string,
   id: string,
   type: string,
-  username?: string,
+  username: string,
   uid?: string,
   url: string,
 }
 
-export type InvitesState = {|
-  pendingInvites: Array<PendingInvite>,
-  acceptedInvites: Array<AcceptedInvite>,
+export type _InvitesState = {|
+  pendingInvites: I.List<PendingInvite>,
+  acceptedInvites: I.List<AcceptedInvite>,
   error: ?Error,
 |}
+export type InvitesState = I.RecordOf<_InvitesState>
 
-export type NotificationsSettingsState = {
+export type _NotificationsSettingsState = {
   name: string,
   subscribed: boolean,
   description: string,
 }
+export type NotificationsSettingsState = I.RecordOf<_NotificationsSettingsState>
 
-export type NotificationsGroupState = {
-  settings: ?Array<NotificationsSettingsState>,
+export type _NotificationsGroupState = {
+  settings: Array<NotificationsSettingsState>,
   unsubscribedFromAll: boolean,
 }
+export type NotificationsGroupState = I.RecordOf<_NotificationsGroupState>
 
-export type NotificationGroups = 'email' | 'app_push' | 'sms'
+export type NotificationSettingsStateGroups = I.Map<string, NotificationsGroupState>
 
-export type NotificationsState = {
+export type _NotificationsState = {
   allowEdit: boolean,
-  groups: {
-    email?: NotificationsGroupState,
-    app_push?: NotificationsGroupState,
-    sms?: NotificationsGroupState,
-  },
+  groups: I.Map<string, NotificationsGroupState>,
```

Comment:
These keys (email, app_push, sms) are coming down from the server and rendered dynamically, so it's better not to hardcode them in the type.